### ISSUE TITLE
wandb loses "global_step" in PyTorch event_writer

### ISSUE THREAD

**Issue by yaroslavvb** [wandb/client#619]
There doesn't seem to be a correspondence between `global_step` and `step` in my graphs. I'm using wandb through eventwriter API in PyTorch. Debugging a bit, it seems

`step` gets converted to `global_step` here
https://github.com/wandb/client/blob/6417dd926abe76dbb7c56e7017d2ee7d1c918eb5/wandb/tensorboard/__init__.py#L205

Then `wandb.log` gets called with this dict, it doesn't see `step` so it assigns one automatically

Custom global step is useful to compare data efficiency consistency across runs -- using "forward calls" as x-axis means your curves look 2x better when doubling batch size or number of workers. Using global data counter for steps gives easier to interpret curves
@vanpelt

**Comment by yaroslavvb** [comment]
For consistency with TensorBoard I think it should keep `step` value as `step` instead of renaming to `global_step` in `tf_summary_to_dict`

**Comment by vanpelt** [comment]
Hey @yaroslavvb we did this intentionally.  To make the graphs lineup you can set the x-axis in the wandb interface to be global_step instead of our step.  There's a global "x-axis" button in the vix panel, and you can set them individually in a custom vis chart.

The reason we did this is that a single tensorboard folder might have multiple tfevent files with different steps such as train, val, or in some cases far more complex setups. We also wanted users to be able to sync tensorboard and call wandb.log for custom objects which detaching the steps helps with.

I know this isn't ideal.  The first thing we should do is in runs where we have a global_step logged we should use that as the default axis.  We're also in the process of refactoring the backend to better support these complex scenarios.

**Comment by yaroslavvb** [comment]
Ah, I see, didn't realize there are now custom X-axis fields. I see "learning rate" and "Train accuracy" in there as well.

I think it's reasonable to assume that most TensorBoard user set their `step` values manually -- the front-end only allows one kind of step and if you don't set it the training accuracy graphs get collapsed into one point

**Comment by vanpelt** [comment]
Yeah, we look for any metrics that are monotonically increasing and make them available as x-axis options.  MLFlow, Tensorboard, and others allow you to go back in time during training and we don't.  By putting the step in metrics we're punting the missing feature by allowing users to log historic steps at the cost of using them as an x-axis (but still being able to export via the python ap).  The big rewrite will enable this behaviour.  We're seeing a bunch of users syncing tensorboard, so we should atleast default to global_step as the x-axis in this case.

**Comment by yaroslavvb** [comment]
Thanks for the info, closing it as duplicate of https://github.com/wandb/client/issues/613
